internal/connectors/plugins/public/modulr/payments.go

```diff
@@ -145,14 +171,39 @@ func transactionToPayment(transaction client.Transaction, from models.PSPAccount
 	return payment, nil
 }
 
+func (p Plugin) fetchAndTranslateTransfer(
+	ctx context.Context,
+	transaction client.Transaction,
+) (*models.PSPPayment, error) {
+	if !transaction.Credit {
+		// Transfer are reprensented as double transactions: one for the source
+		// account and one for the destination account. We don't want to generate
+		// multiple events for the same transfer, and since we are fetching the
+		// whole object, we can safely send it once. Let's ignore the transfer
+		// if the transaction is a debit. It will be fetch on the other side (
+		// the other account's transaction)
```

Comment:
If we're not handling reversals or SEPA direct debit this seems ok, but it could introduce some unexpected behaviour in situations where debtor / creditor become reversed.

Reply:
seems fine based on @paul-nicolas 's explanation about the objects we receive from modulr